# TUGAS 5 (PART 1)
## Evi Ardiyani (G5501211002)
### DATA 2 - PLOT : SCATTER, BAR CHART

In [1]:
using BenchmarkTools
using DataFrames
using DelimitedFiles
using CSV
using Measures
using XLSX
using VegaLite
using VegaDatasets
using Dates

In [3]:
data_kota = XLSX.readdata("Data2.xlsx", #file name
    "kota", #sheet name
    "A1:D16") #cell range
datakota = DataFrame(data_kota,:auto)

16×4 DataFrame
 Row │ x1        x2            x3         x4       
     │ Any       Any           Any        Any      
─────┼─────────────────────────────────────────────
   1 │ Provinsi  Kota          longitude  latitude
   2 │ Kaltim    Samarinda     117.125    -0.625
   3 │ Kaltim    Balikpapan    116.875    -1.125
   4 │ Kaltim    Sendawar      115.625    -0.125
   5 │ Kaltara   Tarakan       117.625    3.375
   6 │ Kaltara   Tanjungselor  117.125    2.875
   7 │ Kaltara   Malinau       116.625    3.375
   8 │ Kalteng   Palangkaraya  113.875    -2.375
   9 │ Kalteng   Purukcahu     114.625    -0.625
  10 │ Kalteng   Pangkalanbun  111.625    -2.625
  11 │ Kalsel    Banjarmasin   114.625    -3.375
  12 │ Kalsel    Kota Baru     116.125    -3.375
  13 │ Kalsel    Barabai       115.375    -2.625
  14 │ Kalbar    Pontianak     109.375    -0.125
  15 │ Kalbar    Ketapang      110.125    -1.875
  16 │ Kalbar    Sintang       111.625    -0.125

In [4]:
@show typeof(datakota)
datakota

typeof(datakota) = DataFrame


16×4 DataFrame
 Row │ x1        x2            x3         x4       
     │ Any       Any           Any        Any      
─────┼─────────────────────────────────────────────
   1 │ Provinsi  Kota          longitude  latitude
   2 │ Kaltim    Samarinda     117.125    -0.625
   3 │ Kaltim    Balikpapan    116.875    -1.125
   4 │ Kaltim    Sendawar      115.625    -0.125
   5 │ Kaltara   Tarakan       117.625    3.375
   6 │ Kaltara   Tanjungselor  117.125    2.875
   7 │ Kaltara   Malinau       116.625    3.375
   8 │ Kalteng   Palangkaraya  113.875    -2.375
   9 │ Kalteng   Purukcahu     114.625    -0.625
  10 │ Kalteng   Pangkalanbun  111.625    -2.625
  11 │ Kalsel    Banjarmasin   114.625    -3.375
  12 │ Kalsel    Kota Baru     116.125    -3.375
  13 │ Kalsel    Barabai       115.375    -2.625
  14 │ Kalbar    Pontianak     109.375    -0.125
  15 │ Kalbar    Ketapang      110.125    -1.875
  16 │ Kalbar    Sintang       111.625    -0.125

### import data curah hujan 15 kota .xlsx

In [5]:
data_curah = XLSX.readdata("Data2.xlsx", #file name
    "curah", #sheet name 
    "A1:P229") #cell range
datacurah = DataFrame(data_curah,:auto)

229×16 DataFrame
 Row │ x1          x2         x3          x4        x5       x6            x7  ⋯
     │ Any         Any        Any         Any       Any      Any           Any ⋯
─────┼──────────────────────────────────────────────────────────────────────────
   1 │ Waktu       Samarinda  Balikpapan  Sendawar  Tarakan  Tanjungselor  Mal ⋯
   2 │ 2001-01-01  162.79     144.95      179.15    84.99    92.16         84.
   3 │ 2001-02-01  133.22     150.68      119.72    38.64    52.84         59.
   4 │ 2001-03-01  82.9774    90.6084     118.192   132.794  120.726       141
   5 │ 2001-04-01  101.92     94.18       119.02    83.98    76.59         110 ⋯
   6 │ 2001-05-01  19.11      16.9        80.44     101.61   77.33         184
   7 │ 2001-06-01  133.49     111.04      83.63     32.17    39.23         46.
   8 │ 2001-07-01  30.99      32.01       44.66     81.03    73.03         87.
   9 │ 2001-08-01  78.8712    93.6209     100.308   71.4065  44.4853       71. ⋯
  10 │ 2001-09-01  157.07     129.06      152.97    94.74    75.48         93.
  11 │ 2001-10-01  151.276    185.813     112.125   61.9074  67.4811       161
  ⋮  │     ⋮           ⋮          ⋮          ⋮         ⋮          ⋮            ⋱
 220 │ 2019-03-01  16.69      31.88       30.76     59.18    41.02         57.
 221 │ 2019-04-01  39.9782    42.5586     98.0969   69.1555  32.43         94. ⋯
 222 │ 2019-05-01  32.5806    29.9935     98.0359   79.4518  49.6385       197
 223 │ 2019-06-01  115.858    117.952     81.2634   84.0763  86.2315       191
 224 │ 2019-07-01  87.85      81.35       71.5      85.04    56.87         104
 225 │ 2019-08-01  36.61      53.57       19.53     44.29    30.97         78. ⋯
 226 │ 2019-09-01  56.5711    41.2557     86.9615   36.5379  25.2525       54.
 227 │ 2019-10-01  102.85     89.3712     195.143   58.71    36.9335       71.
 228 │ 2019-11-01  57.8682    51.2285     98.3349   51.454   51.8301       75.
 229 │ 2019-12-01  189.312    165.989     180.941   88.0717  86.1341       190 ⋯
                                                 10 columns and 208 rows omitted

In [6]:
bulan1=Date.(datacurah[2:121,1])
#membentuk vector{date}

120-element Vector{Date}:
 2001-01-01
 2001-02-01
 2001-03-01
 2001-04-01
 2001-05-01
 2001-06-01
 2001-07-01
 2001-08-01
 2001-09-01
 2001-10-01
 2001-11-01
 2001-12-01
 2002-01-01
 ⋮
 2010-01-01
 2010-02-01
 2010-03-01
 2010-04-01
 2010-05-01
 2010-06-01
 2010-07-01
 2010-08-01
 2010-09-01
 2010-10-01
 2010-11-01
 2010-12-01

In [9]:
using StatsPlots
using Statistics
using Plots
using WebIO
plotlyjs()

Plots.PlotlyJSBackend()

In [10]:
bulan1

120-element Vector{Date}:
 2001-01-01
 2001-02-01
 2001-03-01
 2001-04-01
 2001-05-01
 2001-06-01
 2001-07-01
 2001-08-01
 2001-09-01
 2001-10-01
 2001-11-01
 2001-12-01
 2002-01-01
 ⋮
 2010-01-01
 2010-02-01
 2010-03-01
 2010-04-01
 2010-05-01
 2010-06-01
 2010-07-01
 2010-08-01
 2010-09-01
 2010-10-01
 2010-11-01
 2010-12-01

In [11]:
#mengubah data ke bentuk matriks agar dapat plotting
datamcurah=Matrix(datacurah)
datamcurah=Float64.(datamcurah[2:end,2:end])

228×15 Matrix{Float64}:
 162.79     144.95     179.15    …   58.57         106.28     104.75
 133.22     150.68     119.72        31.77          51.94      76.04
  82.9774    90.6084   118.192       64.7117        96.8544   100.322
 101.92      94.18     119.02        80.39         141.54     213.03
  19.11      16.9       80.44        67.57          48.62     122.97
 133.49     111.04      83.63    …   43.69          62.28      36.73
  30.99      32.01      44.66         4.34          73.2       75.28
  78.8712    93.6209   100.308       -0.000959598    2.92181   29.9689
 157.07     129.06     152.97        45.24         101.82     159.67
 151.276    185.813    112.125      118.654        116.57     129.513
 102.71     107.07     101.83    …   50.23         103.93     109.94
 105.605    105.561    108.623       51.4717       141.545     81.831
  68.66     107.92     167.58        83.91          94.62      94.77
   ⋮                             ⋱                            
  61.4852  

In [12]:
#data kota sendawar, banjarmasin, dan kotabaru
plot(bulan1, datamcurah[1:120,:], size=(1000,500),labels=["Samarinda" "Balikpapan" "Sendawar" "Tarakan" "Tanjungselor" "Malinau" "Palangkaraya" "Purukcahu" "Pangkalanbun" "Banjarmasin" "Kota Baru" "Barabai" "Pontianak" "Ketapang" "Sintang"], title="Data Curah Hujan tahun 2001-2010")

In [13]:
#inisialisasi data Sendawar, Banjarmasin, dan Kotabaru hingga tahun 2010
sendawar=datamcurah[1:120,3]
banjarmasin=datamcurah[1:120,10]
kotabaru=datamcurah[1:120,11]

120-element Vector{Float64}:
 102.649999503046
 117.989999566227
  98.5443661021137
  32.7200000993907
  26.7099998500198
  74.4700004477054
  18.1400001831353
  61.8021675665145
  79.03000000678
 104.108183104418
 114.390000196174
 118.883657439842
 107.47000064142
   ⋮
  27.3999998159707
   2.06000000610948
  12.0099998861551
  23.3600001484156
  74.2900000568479
 121.999999932945
  98.9000001102686
  77.6700002215803
 101.340000441298
 146.970000440255
  56.069999601692
 128.789999805391

In [14]:
#menggabungkan data kota Sendawar, Banjarmasin, dan Kotabaru dalam ssatu matriks
gab=[sendawar banjarmasin kotabaru]

120×3 Matrix{Float64}:
 179.15   151.62    102.65
 119.72   144.16    117.99
 118.192   61.3568   98.5444
 119.02   108.97     32.72
  80.44    25.31     26.71
  83.63    45.45     74.47
  44.66    41.73     18.14
 100.308   47.5789   61.8022
 152.97    97.97     79.03
 112.125   64.1873  104.108
 101.83    72.64    114.39
 108.623   76.7197  118.884
 167.58   346.99    107.47
   ⋮                
 146.87   157.04     27.4
  52.23    63.71      2.06
  53.89    79.31     12.01
 134.79    85.19     23.36
  97.37    74.03     74.29
 126.49    50.96    122.0
 112.68   107.09     98.9
  83.48    85.38     77.67
 109.34    67.12    101.34
 126.74    86.11    146.97
 147.55    78.91     56.07
 115.21    99.1     128.79

In [15]:
#data kota sendawar, banjarmasin, dan kotabaru
plot(bulan1, gab, size=(1000,500),labels=["Sendawar" "Banjarmasin" "Kota Baru"], title="Data Curah Hujan tahun 2001-2010")

In [64]:
sendawar01=maximum(sendawar[1:12]) #sendawar2001
sendawar02=maximum(sendawar[13:24]) #sendawar2002
sendawar03=maximum(sendawar[25:36]) #sendawar2003
sendawar04=maximum(sendawar[37:48]) #sendawar2004
sendawar05=maximum(sendawar[49:60]) #sendawar2005
sendawar06=maximum(sendawar[61:72]) #sendawar2006
sendawar07=maximum(sendawar[73:84]) #sendawar2007
sendawar08=maximum(sendawar[85:96]) #sendawar2008
sendawar09=maximum(sendawar[97:108]) #sendawar2009
sendawar10=maximum(sendawar[109:120]) #sendawar2010
using StatsPlots
tahun=string.(collect(2001:2010))
plot1=bar(tahun,[sendawar01 sendawar02 sendawar03 sendawar04 sendawar05 sendawar06 sendawar07 sendawar08 sendawar09 sendawar10][:],
            #bar_position = :dodge, 
            bar_width=0.7, 
            xticks=(1:10 ,tahun), 
            ylabel="Curah Hujan",
            xlabel="Tahun",
            xrotation=45, 
            label=false, color=:red, title="Curah Hujan Maksimum kota Sendawar tahun 2001-2010",
            size=(700,300))


In [19]:
banjarmasin01=maximum(banjarmasin[1:12]) #banjarmasin2001
banjarmasin02=maximum(banjarmasin[13:24]) #banjarmasin2002
banjarmasin03=maximum(banjarmasin[25:36]) #banjarmasin2003
banjarmasin04=maximum(banjarmasin[37:48]) #banjarmasin2004
banjarmasin05=maximum(banjarmasin[49:60]) #banjarmasin2005
banjarmasin06=maximum(banjarmasin[61:72]) #banjarmasin2006
banjarmasin07=maximum(banjarmasin[73:84]) #banjarmasin2007
banjarmasin08=maximum(banjarmasin[85:96]) #banjarmasin2008
banjarmasin09=maximum(banjarmasin[97:108]) #banjarmasin2009
banjarmasin10=maximum(banjarmasin[109:120]) #banjarmasin2010using StatsPlots
tahun=string.(collect(2001:2010))
plot2=bar(tahun,[banjarmasin01 banjarmasin02 banjarmasin03 banjarmasin04 banjarmasin05 banjarmasin06 banjarmasin07 banjarmasin08 banjarmasin09 banjarmasin09][:],
            #bar_position = :dodge, 
            bar_width=0.7, 
            xticks=(1:10 ,tahun), 
            ylabel="Curah Hujan",
            xlabel="Tahun",
            xrotation=45, 
            label=false, color=:blue, title="Curah Hujan Maksimum kota Banjarmasin tahun 2001-2010",
            size=(700,300))


In [20]:
kotabaru01=maximum(kotabaru[1:12]) #kotabaru2001
kotabaru02=maximum(kotabaru[13:24]) #kotabaru2002
kotabaru03=maximum(kotabaru[25:36]) #kotabaru2003
kotabaru04=maximum(kotabaru[37:48]) #kotabaru2004
kotabaru05=maximum(kotabaru[49:60]) #kotabaru2005
kotabaru06=maximum(kotabaru[61:72]) #kotabaru2006
kotabaru07=maximum(kotabaru[73:84]) #kotabaru2007
kotabaru08=maximum(kotabaru[85:96]) #kotabaru2008
kotabaru09=maximum(kotabaru[97:108]) #kotabaru2009
kotabaru10=maximum(kotabaru[109:120]) #kotabaru2010
using StatsPlots
tahun=string.(collect(2001:2010))
plot3=bar(tahun,[kotabaru01 kotabaru02 kotabaru03 kotabaru04 kotabaru05 kotabaru06 kotabaru07 kotabaru08 kotabaru09 kotabaru10][:],
            #bar_position = :dodge, 
            bar_width=0.7, 
            xticks=(1:10 ,tahun), 
            ylabel="Curah Hujan",
            xlabel="Tahun",
            xrotation=45, 
            label=false, color=:green, title="Curah Hujan kota Maksimum Kotabaru tahun 2001-2010",
            size=(700,300))


# Deskripsi Statistik setiap kota

In [21]:
using StatsBase
println("Deskripsi statistik kota Samarinda : ")
describe(datamcurah[1:end,1])

Deskripsi statistik kota Samarinda : 
Summary Stats:
Length:         228
Missing Count:  0
Mean:           87.058643
Minimum:        1.540000
1st Quartile:   48.827500
Median:         81.240000
3rd Quartile:   122.149999
Maximum:        232.860001
Type:           Float64


In [22]:
println("Deskripsi statistik kota Balikpapan : ")
describe(datamcurah[1:end,2])

Deskripsi statistik kota Balikpapan : 
Summary Stats:
Length:         228
Missing Count:  0
Mean:           84.892471
Minimum:        1.098855
1st Quartile:   48.192500
Median:         82.495000
3rd Quartile:   113.575000
Maximum:        269.680000
Type:           Float64


In [23]:
println("Deskripsi statistik kota Sendawar : ")
describe(datamcurah[1:end,3])

Deskripsi statistik kota Sendawar : 
Summary Stats:
Length:         228
Missing Count:  0
Mean:           109.689979
Minimum:        19.530000
1st Quartile:   71.427500
Median:         98.862871
3rd Quartile:   134.550001
Maximum:        442.270001
Type:           Float64


In [24]:
println("Deskripsi statistik kota Tarakan : ")
describe(datamcurah[1:end,4])

Deskripsi statistik kota Tarakan : 
Summary Stats:
Length:         228
Missing Count:  0
Mean:           69.916848
Minimum:        5.150000
1st Quartile:   49.960000
Median:         68.994999
3rd Quartile:   88.552499
Maximum:        159.440000
Type:           Float64


In [25]:
println("Deskripsi statistik kota Tanjungselor : ")
describe(datamcurah[1:end,5])

Deskripsi statistik kota Tanjungselor : 
Summary Stats:
Length:         228
Missing Count:  0
Mean:           62.431237
Minimum:        10.050000
1st Quartile:   42.712500
Median:         61.335000
3rd Quartile:   76.727499
Maximum:        157.710000
Type:           Float64


In [26]:
println("Deskripsi statistik kota Malinau : ")
describe(datamcurah[1:end,6])

Deskripsi statistik kota Malinau : 
Summary Stats:
Length:         228
Missing Count:  0
Mean:           117.983300
Minimum:        11.400000
1st Quartile:   85.295000
Median:         113.970000
3rd Quartile:   152.045000
Maximum:        249.940000
Type:           Float64


In [27]:
println("Deskripsi statistik kota Palangkaraya : ")
describe(datamcurah[1:end,7])

Deskripsi statistik kota Palangkaraya : 
Summary Stats:
Length:         228
Missing Count:  0
Mean:           120.372809
Minimum:        14.710000
1st Quartile:   78.410000
Median:         116.970000
3rd Quartile:   152.259765
Maximum:        278.930001
Type:           Float64


In [28]:
println("Deskripsi statistik kota Purukcahu : ")
describe(datamcurah[1:end,8])

Deskripsi statistik kota Purukcahu : 
Summary Stats:
Length:         228
Missing Count:  0
Mean:           107.825452
Minimum:        25.440000
1st Quartile:   78.490734
Median:         104.567875
3rd Quartile:   137.802067
Maximum:        212.899998
Type:           Float64


In [29]:
println("Deskripsi statistik kota Banjarmasin : ")
describe(datamcurah[1:end,9])

Deskripsi statistik kota Banjarmasin : 
Summary Stats:
Length:         228
Missing Count:  0
Mean:           85.463823
Minimum:        1.670000
1st Quartile:   53.700000
Median:         79.740000
3rd Quartile:   114.645000
Maximum:        190.040000
Type:           Float64


In [30]:
println("Deskripsi statistik kota Kota Baru : ")
describe(datamcurah[1:end,10])

Deskripsi statistik kota Kota Baru : 
Summary Stats:
Length:         228
Missing Count:  0
Mean:           91.726160
Minimum:        9.390000
1st Quartile:   48.342499
Median:         72.190001
3rd Quartile:   107.090000
Maximum:        440.689999
Type:           Float64


In [31]:
println("Deskripsi statistik kota Barabai : ")
describe(datamcurah[1:end,11])

Deskripsi statistik kota Barabai : 
Summary Stats:
Length:         228
Missing Count:  0
Mean:           64.434233
Minimum:        0.130000
1st Quartile:   35.372570
Median:         57.040000
3rd Quartile:   84.803218
Maximum:        258.269999
Type:           Float64


In [32]:
println("Deskripsi statistik kota Pontianak : ")
describe(datamcurah[1:end,12])

Deskripsi statistik kota Pontianak : 
Summary Stats:
Length:         228
Missing Count:  0
Mean:           93.267343
Minimum:        2.420000
1st Quartile:   57.190000
Median:         83.559999
3rd Quartile:   114.977499
Maximum:        294.168782
Type:           Float64


In [33]:
println("Deskripsi statistik kota Ketapang : ")
describe(datamcurah[1:end,13])

Deskripsi statistik kota Ketapang : 
Summary Stats:
Length:         228
Missing Count:  0
Mean:           62.599796
Minimum:        -0.004263
1st Quartile:   35.497862
Median:         59.545000
3rd Quartile:   84.822500
Maximum:        202.042241
Type:           Float64


In [34]:
println("Deskripsi statistik kota Sintang : ")
describe(datamcurah[1:end,14])

Deskripsi statistik kota Sintang : 
Summary Stats:
Length:         228
Missing Count:  0
Mean:           78.965715
Minimum:        0.000000
1st Quartile:   45.180001
Median:         76.535000
3rd Quartile:   112.365775
Maximum:        213.560001
Type:           Float64


# METODE GUMBEL

In [35]:
#CEK NILAI cs (JIKA >1, METODE GUMBEL PILIHAN TEPAT)

In [36]:
samarinda=datamcurah[1:120,1]
#nilai cs Samarinda
ns=length(samarinda)
meansamarinda = mean(samarinda)
xsamarinda=datamcurah[1:120,1]
ssamarinda=datamcurah[1:120,1]
#nilai sigma
for i in 1:ns
   xsamarinda[i]=(samarinda[i]-meansamarinda).^3
end
#nilai standar deviasi
for i in 1:ns
    ssamarinda[i]=(samarinda[i]-meansamarinda).^2
end
s=sum(ssamarinda)
stdsamarinda=sqrt(s/(ns-1))
a=(ns/((ns-1)*(ns-2)))*sum(xsamarinda)
cssamarinda=a/stdsamarinda.^3

0.6015586712269303

In [37]:
balikpapan=datamcurah[1:120,2]
#nilai cs Balikpapan
ns=length(balikpapan)
meanbalikpapan = mean(balikpapan)
xbalikpapan=datamcurah[1:120,1]
sbalikpapan=datamcurah[1:120,1]
#nilai sigma
for i in 1:ns
   xbalikpapan[i]=(balikpapan[i]-meanbalikpapan).^3
end
#nilai standar deviasi
for i in 1:ns
    sbalikpapan[i]=(balikpapan[i]-meanbalikpapan).^2
end
s=sum(sbalikpapan)
stdbalikpapan=sqrt(s/(ns-1))
a=(ns/((ns-1)*(ns-2)))*sum(xbalikpapan)
csbalikpapan=a/stdbalikpapan.^3

0.6369808501606258

In [38]:
sendawar=datamcurah[1:120,3]
#nilai cs Sendawar
ns=length(sendawar)
meansendawar = mean(sendawar)
xsendawar=datamcurah[1:120,1]
ssendawar=datamcurah[1:120,1]
#nilai sigma
for i in 1:ns
   xsendawar[i]=(sendawar[i]-meansendawar).^3
end
#nilai standar deviasi
for i in 1:ns
    ssendawar[i]=(sendawar[i]-meansendawar).^2
end
s=sum(ssendawar)
stdssendawar=sqrt(s/(ns-1))
a=(ns/((ns-1)*(ns-2)))*sum(xsendawar)
cssendawar=a/stdssendawar.^3

1.9777454929903893

In [39]:
tarakan=datamcurah[1:120,4]
#nilai cs Tarakan
ns=length(tarakan)
meantarakan = mean(tarakan)
xtarakan=datamcurah[1:120,1]
starakan=datamcurah[1:120,1]
#nilai sigma
for i in 1:ns
   xtarakan[i]=(tarakan[i]-meantarakan).^3
end
#nilai standar deviasi
for i in 1:ns
    starakan[i]=(tarakan[i]-meantarakan).^2
end
s=sum(starakan)
stdtarakan=sqrt(s/(ns-1))
a=(ns/((ns-1)*(ns-2)))*sum(xtarakan)
cstarakan=a/stdtarakan.^3


0.32937565458223395

In [40]:
tanjungselor=datamcurah[1:120,5]
#nilai cs Tanjungselor
ns=length(tanjungselor)
meantanjungselor = mean(tanjungselor)
xtanjungselor=datamcurah[1:120,1]
stanjungselor=datamcurah[1:120,1]
#nilai sigma
for i in 1:ns
   xtanjungselor[i]=(tanjungselor[i]-meantanjungselor).^3
end
#nilai standar deviasi
for i in 1:ns
    stanjungselor[i]=(tanjungselor[i]-meantanjungselor).^2
end
s=sum(stanjungselor)
stdtanjungselor=sqrt(s/(ns-1))
a=(ns/((ns-1)*(ns-2)))*sum(xtanjungselor)
cstanjungselor=a/stdtanjungselor.^3

0.7526627391733606

In [41]:
malinau=datamcurah[1:120,6]
#nilai cs Malinau
ns=length(malinau)
meanmalinau = mean(malinau)
xmalinau=datamcurah[1:120,1]
smalinau=datamcurah[1:120,1]
#nilai sigma
for i in 1:ns
   xmalinau[i]=(malinau[i]-meanmalinau).^3
end
#nilai standar deviasi
for i in 1:ns
    smalinau[i]=(malinau[i]-meanmalinau).^2
end
s=sum(smalinau)
stdmalinau=sqrt(s/(ns-1))
a=(ns/((ns-1)*(ns-2)))*sum(xmalinau)
csmalinau=a/stdmalinau.^3

0.4811043703614051

In [42]:
palangkaraya=datamcurah[1:120,7]
#nilai cs Palangkaraya
ns=length(palangkaraya)
meanpalangkaraya = mean(palangkaraya)
xpalangkaraya=datamcurah[1:120,1]
spalangkaraya=datamcurah[1:120,1]
#nilai sigma
for i in 1:ns
   xpalangkaraya[i]=(palangkaraya[i]-meanpalangkaraya).^3
end
#nilai standar deviasi
for i in 1:ns
    spalangkaraya[i]=(palangkaraya[i]-meanpalangkaraya).^2
end
s=sum(spalangkaraya)
stdpalangkaraya=sqrt(s/(ns-1))
a=(ns/((ns-1)*(ns-2)))*sum(xpalangkaraya)
cspalangkaraya=a/stdpalangkaraya.^3


0.4671504803766616

In [43]:
purukcahu=datamcurah[1:120,8]
#nilai cs Purukcahu
ns=length(purukcahu)
meanpurukcahu = mean(purukcahu)
xpurukcahu=datamcurah[1:120,1]
spurukcahu=datamcurah[1:120,1]
#nilai sigma
for i in 1:ns
   xpurukcahu[i]=(purukcahu[i]-meanpurukcahu).^3
end
#nilai standar deviasi
for i in 1:ns
    spurukcahu[i]=(purukcahu[i]-meanpurukcahu).^2
end
s=sum(spurukcahu)
stdpurukcahu=sqrt(s/(ns-1))
a=(ns/((ns-1)*(ns-2)))*sum(xpurukcahu)
cspurukcahu=a/stdpurukcahu.^3


0.1664307404475438

In [44]:
pangkalan=datamcurah[1:120,9]
#nilai cs Pangkalan
ns=length(pangkalan)
meanpangkalan = mean(pangkalan)
xpangkalan=datamcurah[1:120,1]
spangkalan=datamcurah[1:120,1]
#nilai sigma
for i in 1:ns
   xpangkalan[i]=(pangkalan[i]-meanpangkalan).^3
end
#nilai standar deviasi
for i in 1:ns
    spangkalan[i]=(pangkalan[i]-meanpangkalan).^2
end
s=sum(spangkalan)
stdpangkalan=sqrt(s/(ns-1))
a=(ns/((ns-1)*(ns-2)))*sum(xpangkalan)
cspangkalan=a/stdpangkalan.^3

0.18260601970084617

In [45]:
banjarmasin=datamcurah[1:120,10]
#nilai cs Banjarmasin
ns=length(banjarmasin)
meanbanjarmasin = mean(banjarmasin)
xbanjarmasin=datamcurah[1:120,1]
sbanjarmasin=datamcurah[1:120,1]
#nilai sigma
for i in 1:ns
   xbanjarmasin[i]=(banjarmasin[i]-meanbanjarmasin).^3
end
#nilai standar deviasi
for i in 1:ns
    sbanjarmasin[i]=(banjarmasin[i]-meanbanjarmasin).^2
end
s=sum(sbanjarmasin)
stdbanjarmasin=sqrt(s/(ns-1))
a=(ns/((ns-1)*(ns-2)))*sum(xbanjarmasin)
csbanjarmasin=a/stdbanjarmasin.^3

2.2863467007343794

In [46]:
kotabaru=datamcurah[1:120,11]
#nilai cs Kota Baru
ns=length(banjarmasin)
meankotabaru = mean(kotabaru)
xkotabaru=datamcurah[1:120,1]
skotabaru=datamcurah[1:120,1]
#nilai sigma
for i in 1:ns
   xkotabaru[i]=(kotabaru[i]-meankotabaru).^3
end
#nilai standar deviasi
for i in 1:ns
    skotabaru[i]=(kotabaru[i]-meankotabaru).^2
end
s=sum(skotabaru)
stdkotabaru=sqrt(s/(ns-1))
a=(ns/((ns-1)*(ns-2)))*sum(xkotabaru)
cskotabaru=a/stdkotabaru.^3

1.347776727961134

In [47]:
barabai=datamcurah[1:120,12]
#nilai cs Barabai
ns=length(barabai)
meanbarabai = mean(barabai)
xbarabai=datamcurah[1:120,1]
sbarabai=datamcurah[1:120,1]
#nilai sigma
for i in 1:ns
   xbarabai[i]=(barabai[i]-meanbarabai).^3
end
#nilai standar deviasi
for i in 1:ns
    sbarabai[i]=(barabai[i]-meanbarabai).^2
end
s=sum(sbarabai)
stdbarabai=sqrt(s/(ns-1))
a=(ns/((ns-1)*(ns-2)))*sum(xbarabai)
csbarabai=a/stdbarabai.^3

1.1233202369823374

In [48]:
pontianak=datamcurah[1:120,13]
#nilai cs Pontianak
ns=length(pontianak)
meanpontianak = mean(pontianak)
xpontianak=datamcurah[1:120,1]
spontianak=datamcurah[1:120,1]
#nilai sigma
for i in 1:ns
   xpontianak[i]=(pontianak[i]-meanpontianak).^3
end
#nilai standar deviasi
for i in 1:ns
    spontianak[i]=(pontianak[i]-meanpontianak).^2
end
s=sum(spontianak)
stdpontianak=sqrt(s/(ns-1))
a=(ns/((ns-1)*(ns-2)))*sum(xpontianak)
cspontianak=a/stdpontianak.^3

0.7780796893984784

In [49]:
ketapang=datamcurah[1:120,14]
#nilai cs Ketapang
ns=length(ketapang)
meanketapang = mean(ketapang)
xketapang=datamcurah[1:120,1]
sketapang=datamcurah[1:120,1]
#nilai sigma
for i in 1:ns
   xketapang[i]=(ketapang[i]-meanketapang).^3
end
#nilai standar deviasi
for i in 1:ns
    sketapang[i]=(ketapang[i]-meanketapang).^2
end
s=sum(sketapang)
stdketapang=sqrt(s/(ns-1))
a=(ns/((ns-1)*(ns-2)))*sum(xketapang)
csketapang=a/stdketapang.^3

0.16100342640718684

In [50]:
sintang=datamcurah[1:120,6]
#nilai cs Sintang
ns=length(sintang)
meansintang = mean(sintang)
xsintang=datamcurah[1:120,1]
ssintang=datamcurah[1:120,1]
#nilai sigma
for i in 1:ns
   xsintang[i]=(sintang[i]-meansintang).^3
end
#nilai standar deviasi
for i in 1:ns
    ssintang[i]=(sintang[i]-meansintang).^2
end
s=sum(ssintang)
stdsintang=sqrt(s/(ns-1))
a=(ns/((ns-1)*(ns-2)))*sum(xsintang)
cssintang=a/stdsintang.^3

0.4811043703614051

In [68]:
@show cssamarinda
@show csbalikpapan
@show cssendawar
@show cstarakan
@show cstanjungselor
@show csmalinau
@show cspalangkaraya
@show cspurukcahu
@show cspangkalan
@show csbanjarmasin
@show cskotabaru
@show csbarabai
@show cspontianak
@show csketapang
@show cssintang

cssamarinda = 0.6015586712269303
csbalikpapan = 0.6369808501606258
cssendawar = 1.9777454929903893
cstarakan = 0.32937565458223395
cstanjungselor = 0.7526627391733606
csmalinau = 0.4811043703614051
cspalangkaraya = 0.4671504803766616
cspurukcahu = 0.1664307404475438
cspangkalan = 0.18260601970084617
csbanjarmasin = 2.2627254706369326
cskotabaru = 1.347776727961134
csbarabai = 1.1233202369823374
cspontianak = 0.7780796893984784
csketapang = 0.16100342640718684
cssintang = 0.4811043703614051


0.4811043703614051

In [70]:
cs=[cssamarinda csbalikpapan cssendawar cstarakan cstanjungselor csmalinau cspalangkaraya cspurukcahu cspangkalan csbanjarmasin cskotabaru csbarabai cspontianak csketapang cssintang]
kota=["Samarinda" ; "Balikpapan"; "Sendawar" ;"Tarakan" ;"Tanjungselor" ; "Malinau" ;"Palangkaraya" ;"Purukcahu" ;"Pangkalanbun" ;"Banjarmasin" ;"Kota Baru"; "Barabai"; "Pontianak" ;"Ketapang" ;"Sintang"]
bar(kota,[cssamarinda csbalikpapan cssendawar cstarakan cstanjungselor csmalinau cspalangkaraya cspurukcahu cspangkalan csbanjarmasin cskotabaru csbarabai cspontianak csketapang cssintang][:],
            #bar_position = :dodge, 
            bar_width=0.7, 
            xticks=(0.5:15 ,kota), 
            ylabel="Skewness",
            #xlabel="Kota",
            xrotation=90, 
            label=false, color=:green, title="Skewness Value",
            size=(500,300))

Nilai Cs masing-masing kota adalah : 
<br>Sendawar : 1.589410126854073
<br>Banjarmasin : 2.2627254706369326
<br>Kota Baru : 1.5044588607806408
<br> Cs>1 maka,
<br> <b> Metode Gumbel untuk ketiga kota dapat diterapkan<b/>

CURAH HUJAN RENCANA KOTA : <b>SENDAWAR</b>

In [71]:
#menentukan nilai b
#lihat tabel Yn =0.4952 dan Sn=0.9496
b=(meansendawar-((0.4952*stdssendawar)/0.9496))
println("b=$b")
a=stdssendawar/0.9496
println("a=$a")
yt=[0.366513 1.49994 2.250367 2.673752 2.970195 3.198534 3.901939 4.600149]
xtsendawar=b.+a.*yt
println("curah hujan rencana periode T : $xtsendawar")

b=86.34447942884327
a=59.26802951358477
curah hujan rencana periode T : [108.06698272995577 175.2429676174496 219.7192972012405 244.81249187684958 262.3820843499452 275.9152869410476 317.6047152410507 358.98624612773074]


CURAH HUJAN RENCANA KOTA : <b>BANJARMASIN</b>

In [72]:
#menentukan nilai b
#lihat tabel Yn =0.4952 dan Sn=0.9496
b=(meanbanjarmasin-((0.4952*stdbanjarmasin)/0.9496))
println("b=$b")
a=stdbanjarmasin/0.9496
println("a=$a")
yt=[0.366513 1.49994 2.250367 2.673752 2.970195 3.198534 3.901939 4.600149]
xtbanjarmasin=b.+a.*yt
println("curah hujan rencana periode T : $xtbanjarmasin")

b=54.53373745572072
a=75.1058613822895
curah hujan rencana periode T : [82.06101202852778 167.18802317747205 223.5494894169994 255.34818453834004 277.6127914040901 294.7623886862607 347.59222711187005 400.0318905875984]


CURAH HUJAN RENCANA KOTA : <b>KOTA BARU</b>

In [73]:
#menentukan nilai b
#lihat tabel Yn =0.4952 dan Sn=0.9496
b=(meankotabaru-((0.4952*stdkotabaru)/0.9496))
println("b=$b")
a=stdkotabaru/0.9496
println("a=$a")
yt=[0.366513 1.49994 2.250367 2.673752 2.970195 3.198534 3.901939 4.600149]
xtkotabaru=b.+a.*yt
println("curah hujan rencana periode T : $xtkotabaru")

b=46.72746449835119
a=43.30303213929903
curah hujan rencana periode T : [62.59858871682209 111.67941452537139 144.1751790245691 162.50903328686624 175.34591404333648 185.23368509899188 215.69325442093552 245.92786449091548]


In [74]:
periode=[2 5 10 15 20 25 50 100]
periode=string.(periode)

1×8 Matrix{String}:
 "2"  "5"  "10"  "15"  "20"  "25"  "50"  "100"

In [75]:
bar(periode,xtsendawar, 
            bar_width=0.7, 
            title="Curah Hujan Rencana Maximum Kota Sendawar",
            ylabel="Curah Hujan Maximum",
            xlabel="Periode t-Tahun",
            color=:blue,
            alpha=0.7,
            xrotation=45, 
            size=(700,300),
            legend=false)

In [76]:
bar(periode,xtbanjarmasin, 
            bar_width=0.7, 
            title="Curah Hujan Rencana Maximum Kota Banjarmasin",
            ylabel="Curah Hujan Maximum",
            xlabel="Periode t-Tahun",
            color=:red,
            alpha=0.7,
            xrotation=45, 
            size=(700,300),
            legend=false)

In [77]:
bar(periode,xtkotabaru, 
            bar_width=0.7, 
            title="Curah Hujan Rencana Maximum Kota Kota Baru",
            ylabel="Curah Hujan Maximum",
            xlabel="Periode t-Tahun",
            color=:purple,
            alpha=0.7,
            xrotation=45, 
            size=(700,300),
            legend=false)

In [78]:
p1 = bar(periode,xtsendawar, label=false, color=:blue,title="CURAH HUJAN RENCANA KOTA SENDAWAR PERIODE-T",titlefontsize=8)#,xlabel="Periode-T tahun")
p2 = bar(periode,xtbanjarmasin,label=false, color=:red,title="CURAH HUJAN RENCANA KOTA BANJARMASIN PERIODE-T",titlefontsize=8)#,xlabel="Periode-T tahun")
p3 = bar(periode,xtkotabaru,label=false, color=:purple,title="CURAH HUJAN RENCANA KOTA KOTABARU PERIODE-T",titlefontsize=8)#,xlabel="Periode-T tahun")
plot(p1, p2, p3, layout = (3, 1), xrotation=45, bottom_margin=0mm,)

In [79]:
#inisialisasi data Sendawar, Banjarmasin, dan Kotabaru hingga tahun 2010
sendawarg=datamcurah[1:228,3]
banjarmasing=datamcurah[1:228,10]
kotabarug=datamcurah[1:228,11]
sendawar11=maximum(sendawarg[121:132]) #sendawar2001
sendawar12=maximum(sendawarg[133:144]) #sendawar2002
sendawar13=maximum(sendawarg[145:156]) #sendawar2003
sendawar14=maximum(sendawarg[157:168]) #sendawar2004
sendawar15=maximum(sendawarg[169:180]) #sendawar2005
sendawar16=maximum(sendawarg[181:192]) #sendawar2006
sendawar17=maximum(sendawarg[193:204]) #sendawar2007
sendawar18=maximum(sendawarg[205:216]) #sendawar2008
sendawar19=maximum(sendawarg[217:228]) #sendawar2009
tahun2=string.(collect(2011:2019))
plot4=bar(tahun2,[sendawar11 sendawar12 sendawar13 sendawar14 sendawar15 sendawar16 sendawar17 sendawar18 sendawar19][:],
            #bar_position = :dodge, 
            bar_width=0.7, 
            xticks=(1:9 ,tahun2), 
            ylabel="Curah Hujan",
            xlabel="Tahun",
            xrotation=45, 
            label=false, color=:blue, title="Curah Hujan kota Sendawar tahun 2011-2019",
            size=(500,300))


In [80]:
banjarmasin11=maximum(banjarmasing[121:132]) #banjarmasin2001
banjarmasin12=maximum(banjarmasing[133:144]) #banjarmasin2002
banjarmasin13=maximum(banjarmasing[145:156]) #banjarmasin2003
banjarmasin14=maximum(banjarmasing[157:168]) #banjarmasin2004
banjarmasin15=maximum(banjarmasing[169:180]) #banjarmasin2005
banjarmasin16=maximum(banjarmasing[181:192]) #banjarmasin2006
banjarmasin17=maximum(banjarmasing[193:204]) #banjarmasin2007
banjarmasin18=maximum(banjarmasing[205:216]) #banjarmasin2008
banjarmasin19=maximum(banjarmasing[217:228]) #banjarmasin2009
tahun2=string.(collect(2011:2019))
plot4=bar(tahun2,[banjarmasin11 banjarmasin12 banjarmasin13 banjarmasin14 banjarmasin15 banjarmasin16 banjarmasin17 banjarmasin18 banjarmasin19][:],
            #bar_position = :dodge, 
            bar_width=0.7, 
            xticks=(1:9 ,tahun2), 
            ylabel="Curah Hujan",
            xlabel="Tahun",
            xrotation=45, 
            label=false, color=:red, title="Curah Hujan kota Banjarmasin tahun 2011-2019",
            size=(500,300))


In [81]:
kotabaru11=maximum(kotabarug[121:132]) #kotabaru2001
kotabaru12=maximum(kotabarug[133:144]) #kotabaru2002
kotabaru13=maximum(kotabarug[145:156]) #kotabaru2003
kotabaru14=maximum(kotabarug[157:168]) #kotabaru2004
kotabaru15=maximum(kotabarug[169:180]) #kotabaru2005
kotabaru16=maximum(kotabarug[181:192]) #kotabaru2006
kotabaru17=maximum(kotabarug[193:204]) #kotabaru2007
kotabaru18=maximum(kotabarug[205:216]) #kotabaru2008
kotabaru19=maximum(kotabarug[217:228]) #kotabaru2009
tahun2=string.(collect(2011:2019))
plot4=bar(tahun2,[kotabaru11 kotabaru12 kotabaru13 kotabaru14 kotabaru15 kotabaru16 kotabaru17 kotabaru18 kotabaru19][:],
            #bar_position = :dodge, 
            bar_width=0.7, 
            xticks=(1:9 ,tahun2), 
            ylabel="Curah Hujan",
            xlabel="Tahun",
            xrotation=45, 
            label=false, color=:purple, title="Curah Hujan kota Kotabaru tahun 2011-2019",
            size=(500,300))
